add article about webpack optimization (#235)
M6Web/m6web.github.io: _posts/2020-03-05-hunting-webpack-performances.md

---
layout: post

> title: "How did I cut the time to webpack build for my project in half?"

Reviewer: ```suggestion
title: "How did we cut in half the time for webpack to build?"
```

It took me about 10 minutes because there were few breaking changes and I **gained more than 1min30** on the build time.
The sadness of telling me that I had lost a morning was quickly replaced by the joy of having managed to reduce the build time by a quarter.

### IgnorePlugin, TerserPlugin

- `TerserPlugin` is used to uglify the javascript code in order to reduce its size and readability. It's a relatively long process, but *39 seconds* is too much.
Just by updating the version of TerserPlugin to use the one integrated in Webpack, I could **reduce by 20 seconds** the build time.

- `IgnorePlugin` is a core plugin that was used a lot in our application to avoid loading certain scripts in order to reduce the weight of the site.
It was necessary, but today with Webpack we can use much better than that. Dynamic Import, ContextReplacement, there are plenty of solutions.
As a general rule, one should avoid compiling files and then not using them.

### Recommendations from the community

To improve the build perfs webpack provides a web page listing the actions to take to hunt what takes time.

> Je vous recommande fortement de vous y intéresser.

Reviewer 2: Is this frend sentence really wanted? 🤔
